### PR TITLE
tidb: Split datasource name to uri and dbName

### PR DESCRIPTION
**Pull request by shenli**
The DBName may be different from data path name.

### CODE REVIEW

File: driver.go [hunk 1]
```diff
@@ -131,22 +131,30 @@ func (d *sqlDriver) lock() func() {
...
 		return nil, errors.Trace(err)
 	}
 
 	driver := &sqlDriver{}
 	switch {
-	case strings.HasPrefix(name, "file://"):
-		name = name[len("file://"):]
-	case strings.HasPrefix(name, "memory://"):
-		name = name[len("memory://"):]
+	case strings.HasPrefix(uri, "file://"):
```
Comment (siddontang) [inline driver.go]: does any engine use `file://` as the prefix ?

File: driver.go [hunk 2]
```diff
@@ -131,22 +131,30 @@ func (d *sqlDriver) lock() func() {
 // efficient re-use.
 //
 // The returned connection is only used by one goroutine at a time.
-func (d *sqlDriver) Open(name string) (driver.Conn, error) {
-	store, err := NewStore(name)
+func (d *sqlDriver) Open(dataSource string) (driver.Conn, error) {
+	// Split the dataSource to uri and dbName
+	i := strings.LastIndex(dataSource, "/")
+	if i == -1 {
+		return nil, errors.Errorf("Wrong format of datasource string: %q", dataSource)
```
Comment (qiuyesuifeng) [inline driver.go]: Wrong -> Bad?

File: driver.go [hunk 3]
```diff
@@ -131,22 +131,30 @@ func (d *sqlDriver) lock() func() {
...
 	driver := &sqlDriver{}
 	switch {
-	case strings.HasPrefix(name, "file://"):
-		name = name[len("file://"):]
-	case strings.HasPrefix(name, "memory://"):
-		name = name[len("memory://"):]
+	case strings.HasPrefix(uri, "file://"):
+		uri = uri[len("file://"):]
+	case strings.HasPrefix(uri, "memory://"):
+		uri = uri[len("memory://"):]
```
Comment (qiuyesuifeng) [inline driver.go]: need use default to return an error?

File: driver.go (hunk 1 above)
Comment (shenli) [inline driver.go]: The switch statement is useless now. I will remove it.

**Comment by shenli** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**